src/meetingreadinesschecker/DefaultMeetingReadinessChecker.ts

```diff
@@ -0,0 +1,451 @@
+      }
+    };
+    try {
+      const permissionResult = await audioVideo.chooseAudioInputDevice(audioInputDeviceInfo);
+      if (
+        permissionResult === DevicePermission.PermissionDeniedByBrowser ||
+        permissionResult === DevicePermission.PermissionDeniedByUser
+      ) {
+        return CheckAudioConnectivityFeedback.AudioInputPermissionDenied;
+      }
+    } catch (error) {
+      this.logger.error(
+        `MeetingReadinessChecker: Failed to get audio input device with error ${error}`
+      );
+      return CheckAudioConnectivityFeedback.FailureToGetAudioInput;
+    }
+    audioVideo.realtimeSubscribeToAttendeeIdPresence(attendeePresenceHandler);
+    if (!(await this.startMeeting())) {
+      audioVideo.realtimeUnsubscribeToAttendeeIdPresence(attendeePresenceHandler);
+      return CheckAudioConnectivityFeedback.ConnectionFailure;
+    }
+    await this.executeTimeoutTask(async () => {
+      return audioPresence;
+    });
+    audioVideo.realtimeUnsubscribeToAttendeeIdPresence(attendeePresenceHandler);
+    await this.stopMeeting();
+    return audioPresence
+      ? CheckAudioConnectivityFeedback.Success
+      : CheckAudioConnectivityFeedback.NoAudioPresence;
+  }
+
+  async checkVideoConnectivity(
+    videoInputDeviceInfo: MediaDeviceInfo
+  ): Promise<CheckVideoConnectivityFeedback> {
+    const audioVideo = this.meetingSession.audioVideo;
+
+    try {
+      const permissionResult = await audioVideo.chooseVideoInputDevice(videoInputDeviceInfo);
+      if (
+        permissionResult === DevicePermission.PermissionDeniedByBrowser ||
+        permissionResult === DevicePermission.PermissionDeniedByUser
+      ) {
+        return CheckVideoConnectivityFeedback.VideoInputPermissionDenied;
+      }
+    } catch (error) {
+      this.logger.error(
+        `MeetingReadinessChecker: Failed to get video input device with error ${error}`
+      );
+      return CheckVideoConnectivityFeedback.FailureToGetVideoInput;
+    }
+
+    if (!(await this.startMeeting())) {
+      return CheckVideoConnectivityFeedback.ConnectionFailure;
+    }
+
+    let packetsSent = 0;
+    audioVideo.startLocalVideoTile();
+    await this.executeTimeoutTask(async () => {
+      const rawStats = await audioVideo.getRTCPeerConnectionStats();
+      if (rawStats) {
+        rawStats.forEach(report => {
+          if (report.type === 'outbound-rtp' && report.mediaType === 'video') {
+            packetsSent = report.packetsSent;
+          }
+        });
+      }
+      return packetsSent > 0;
+    });
+    audioVideo.stopLocalVideoTile();
+    await this.stopMeeting();
+    if (packetsSent <= 0) {
+      return CheckVideoConnectivityFeedback.NoPackageSentToServer;
+    }
+    return CheckVideoConnectivityFeedback.Success;
+  }
+
+  async checkNetworkUDPConnectivity(): Promise<CheckNetworkUDPConnectivityFeedback> {
+    try {
+      this.originalURLRewriter = this.meetingSession.configuration.urls.urlRewriter;
+    } catch (e) {
```

Comment:
Two minor comments:
- `(e)` => `(error)`
- This checker's constructor takes in the "required" `meetingSession` parameter, so this "catch" block seems unnecessary. (The builder can pass `undefined` there. Then, you need to add the same meeting session existence check in other checker APIs too.)

Reply:
See https://github.com/aws/amazon-chime-sdk-js/pull/649#discussion_r483166541 for the second bullet point.